Core/include/Acts/Utilities/Any.hpp

```diff
@@ -0,0 +1,427 @@
+    if (m_handler->typeSize <= SIZE) {
+      T* ptr = reinterpret_cast<T*>(m_data.data());
+      _ACTS_ANY_DEBUG("As local: " << ptr);
+      return *ptr;
+    } else {
+      // is heap allocated
+      void* ptr = *reinterpret_cast<void**>(m_data.data());
+      _ACTS_ANY_DEBUG("As heap: " << ptr);
+      return *reinterpret_cast<T*>(ptr);
+    }
+  }
+
+  template <typename T>
+  const T& as() const {
+    static_assert(std::is_same_v<T, std::decay_t<T>>,
+                  "Please pass the raw type, no const or ref");
+    assert(makeHandler<T>() == m_handler && "Bad type access");
+    if (m_handler->typeSize <= SIZE) {
+      const T* ptr = reinterpret_cast<const T*>(m_data.data());
+      _ACTS_ANY_DEBUG("As local: " << ptr);
+      return *ptr;
+    } else {
+      // is heap allocated
+      const void* ptr = *reinterpret_cast<void* const*>(m_data.data());
+      _ACTS_ANY_DEBUG("As heap: " << ptr);
+      return *reinterpret_cast<const T*>(ptr);
+    }
+  }
+
+  ~AnyBase() { destroy(); }
+
+  AnyBase(const AnyBase& other) {
+    if (m_handler == nullptr) {  // this object is empty
+      m_handler = other.m_handler;
+      assert(m_handler != nullptr && "Handler is null");
+      copyConstruct(other);
+    } else {
+      // @TODO: Support assigning between different types
+      assert(m_handler == other.m_handler);
+      copy(other);
+    }
+  }
+
+  AnyBase& operator=(const AnyBase& other) {
+    if (m_handler == nullptr) {  // this object is empty
+      m_handler = other.m_handler;
+      assert(m_handler && "Handler is null");
+      copyConstruct(other);
+    } else {
+      // @TODO: Support assigning between different types
+      assert(m_handler == other.m_handler);
+      copy(other);
+    }
+    return *this;
+  }
+
+  AnyBase(AnyBase&& other) {
+    if (m_handler == nullptr) {  // this object is empty
+      m_handler = other.m_handler;
+      assert(m_handler != nullptr && "Handler is null");
+      moveConstruct(std::move(other));
+    } else {
+      // @TODO: Support assigning between different types
+      assert(m_handler == other.m_handler);
+      move(std::move(other));
+    }
+  }
+
+  AnyBase& operator=(AnyBase&& other) {
+    if (m_handler == nullptr) {  // this object is empty
+      m_handler = other.m_handler;
+      assert(m_handler && "Handler is null");
+      moveConstruct(std::move(other));
+    } else {
+      // @TODO: Support assigning between different types
+      assert(m_handler == other.m_handler);
+      move(std::move(other));
+    }
+    return *this;
+  }
+
+  operator bool() const { return m_handler != nullptr; }
+
+ private:
+  struct Handler {
+    void (*destroy)(void* ptr) = nullptr;
+    void (*moveConstruct)(void* from, void* to) = nullptr;
+    void (*move)(void* from, void* to) = nullptr;
+    void (*copyConstruct)(const void* from, void* to) = nullptr;
+    void (*copy)(const void* from, void* to) = nullptr;
+    std::size_t typeSize{0};
```

Comment:
You're only ever using this to test if you're stored on the stack or the heap, so you might as well use a boolean that says just that to reduce runtime comparisons and hopefully slightly improve performance.

Reply:
Fair enough.